kmod/patch/livepatch-patch-hook.c

```diff
@@ -0,0 +1,324 @@
+/*
+ * Copyright (C) 2013-2014 Josh Poimboeuf <jpoimboe@redhat.com>
+ * Copyright (C) 2014 Seth Jennings <sjenning@redhat.com> 
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA,
+ * 02110-1301, USA.
+ */
+
+#define pr_fmt(fmt) KBUILD_MODNAME ": " fmt
+
+#include <linux/kernel.h>
+#include <linux/module.h>
+#include <linux/list.h>
+#include <linux/slab.h>
+
+#include <linux/livepatch.h>
+
+#include "kpatch-patch.h"
+
+struct klp_patch *patch;
```

Comment:
Because of the scaffolding, there are three different structure "languages" being used: kpatch-build, scaffolding, and klp.  It kind of threw me for a loop when trying to read this code.  It would be easier to follow if the same variable, struct, and function naming conventions were used consistently throughout the entire file.  Something like:

- for kpatch-build: kpatch/kobject/kreloc/kfunc
- for temporary scaffolding: tpatch/tobject/treloc/tfunc (or spatch/sobject/sreloc/sfunc)
- for klp: lpatch/lobject/lreloc/tfunc

And also maybe a comment describing the fact that there are three languages and the different naming conventions.